src/ray/common/scheduling/scheduling_ids.h

```diff
@@ -37,6 +37,7 @@ enum PredefinedResourcesEnum {
   MEM,
   GPU,
   OBJECT_STORE_MEM,
+  NODE_WORKERS,
```

Comment:
i wonder if we could discuss about the alternative approach where we track leased worker size directly? Rather than adding workers as a resource. I am a bit concern of the implications here since it's technically not a resource. 

But I do see the benefits of how easy it is to integrate workers into idle definition with this current approach.

Reply:
I guess if we decide to take that approach it is fine, but feels like more code and more special casing to track. I guess for instance Linux can have virtual devices that don't map to hardware

Reply:
agree with ricky here. fundamentally the workers are implementation detail rather than a logical resource that a client can schedule against.

Reply:
cc @jjyao - could you comment on how this might impact the schedling?

I see we already added implicit resources, i wonder if adding workers as resources internally is also fine.

Reply:
I actually just changed it in the latest revision so that the resource idle time map keys on a union (std::variant) of types instead of just ResourceID. This way we can avoid accidentally incorporating "scheduling/availability semantics" into a resource type that only has "reporting the presence of work semantics"

Happy to change it back, it's quick to do either way and I don't have a strong preference since neither is a very invasive approach.

Reply:
Yeah, I see. I am kind of liking the worker as resources better than the variant thing in terms of readability actually. 

Or do you think it will be a good alternative to just a `leased_workers_active_time` in the local resource manager or something alongside to record the leased worker time?

Reply:
I would prefer not to have a separate `leased_workers_active_time` if it can be streamlined into the current codepath. As to before/after, maybe we can get a quick consensus with @jjyao and @scv119?